stan/math/prim/mat/prob/ordered_logistic_lpmf.hpp

```diff
@@ -46,55 +51,196 @@ namespace stan {
      * @param c Positive increasing vector of cutpoints.
      * @return Log probability of outcome given location and
      * cutpoints.
+     * @throw std::domain_error If the outcome is not between 1 and
+     * the number of cutpoints plus 2; if the cutpoint vector is
+     * empty; if the cutpoint vector contains a non-positive,
+     * non-finite value; or if the cutpoint vector is not sorted in
+     * ascending order.
+     */
+    template <bool propto, typename T_loc, typename T_cut>
+    typename return_type<T_loc>::type
+    ordered_logistic_lpmf(int y, const T_loc& lambda,
+                          const Eigen::Matrix<T_cut, Eigen::Dynamic, 1>& c) {
+      typename return_type<T_loc>::type logp_n(0.0);
+
+      using std::exp;
+      using std::log;
 
+      static const std::string function = "ordered_logistic";
+
+      int K = c.size() + 1;
+
+      check_bounded(function, "Random variable", y, 1, K);
+      check_finite(function, "Location parameter", lambda);
+      check_greater(function, "Size of cut points parameter", c.size(), 0);
+      check_ordered(function, "Cut-points", c);
+      check_finite(function, "Final cut-point", c(c.size()-1));
+      check_finite(function, "First cut-point", c(0));
+
+      if (y == 1) {
```

Comment:
You could've used the vector view pattern on the final argument to vectorize.  Not sure if it'd work for the integer arguments.

For this particular case, you could've avoided the local `logp_n` and just returned the results directly.  It's more efficient than allocating and incrementing for autodiff types.

Anyway, we can clean this up later, so no need to fix for this request.